buildroot/packages/configd/configd.c

```diff
@@ -0,0 +1,201 @@
+  return jobj;
+}
+
+// write config to click filesystem
+int write_config(struct json_object *json) {
+
+  // get all top-level keys
+  json_object_object_foreach(json, key, value) {
+
+    // for the "ports" key
+    if (strcmp(key, "ports") == 0) {
+      // get all ports
+      json_object_object_foreach(value, port, portconfig) {
+        // printf("%s  -> %s\n", port, json_object_get_string(config));
+
+        // get config for a port
+        json_object_object_foreach(portconfig, item, itemconfig) {
+        //   printf("%s  -> %s\n", item, json_object_get_string(itemconfig));
+
+          // for the "enabled" key
+          if (strcmp(item, "enabled") == 0) {
+            bool enabled = json_object_get_boolean(itemconfig);
+            const char* state = get_field(read_switch_port_table("dump_port_phy_cfgs", atoi(port)), 2);
+
+            if (enabled && (strcmp(state, "off") == 0)) {
+              printf("port %s enabled", port);
+              char command[40];
+              snprintf(command, 40, "PORT %s, MODE aneg", port);
+              write_switch_port_table("set_port_phy_cfgs", command);
+            } else if (!enabled && (strcmp(state, "aneg") == 0)) {
+              printf("port %s disabled", port);
+              char command[40];
+              snprintf(command, 40, "PORT %s, MODE off", port);
+              write_switch_port_table("set_port_phy_cfgs", command);
+            }
+          }
+          
+          // for the "poe" key
+          if (poe_capable && strcmp(item, "poe") == 0) {
+            // get poe config
+            json_object_object_foreach(itemconfig, poeitem, poeitemconfig) {
+
+              // for the poe "enabled" key
+              if (strcmp(poeitem, "enabled") == 0) {
+                bool enabled = json_object_get_boolean(poeitemconfig);
+                bool state = port_state(&pd690xx, atoi(port));
+
+                if (enabled && (state == PORT_DISABLED)) {
+                  port_enable(&pd690xx, atoi(port));
+                } else if (!enabled && (state == PORT_ENABLED)) {
+                  port_disable(&pd690xx, atoi(port));
+                }
+              }
+
+            }
+          }
+
+        }
+
+      }
+    }
+
+  }
+}
+
+// poll config file for changes
+void poll() {
+  time_t mtime;
+  time(&mtime);
+  struct stat st;
+  struct json_object *current = read_config();
+  struct json_object *modified;
+
+  int file = open(CONFIG_FILE, O_RDONLY);
+  if(file == -1) { 
+    printf("unable to open file\n"); 
+    return; 
+  } 
+
+  while(true) {
+    // get file modification time
+    if(fstat(file, &st))   { 
+      printf("fstat error: [%s]\n", strerror(0)); 
+      close(file); 
+      return;
+    }
+
+    if (difftime(st.st_mtime, mtime) > 0) {
+      mtime = st.st_mtime;
+      modified = json_object_from_file(CONFIG_FILE);
+      if (!json_object_equal(current, modified)) {
+        current = modified;
+        write_config(modified);
+      }
+    }
+
+    sleep(1);
```

Comment:
A refresh every second is probably excessive, I would default to something like 10 seconds and allow the user to override the polling interval with an argument (e.g. `--poll 5`)

Reply:
That's fair. I had it at 10 but probably reduced it while testing.

Any chance we can add inotify functionality to the device? Maybe we'd still want both options (not aware if there's an upside to polling if it's not necessary).

Reply:
Added. I only implemented the short-form option (`-p`) for consistency with the other 2 packages you've already written (and also because I forgot until re-reading your comment just now :sweat_smile:) but can add the long-form as well, if you'd like.